sound/soc/sof/intel/hda-dai.c

```diff
@@ -243,7 +239,8 @@ static int hda_link_dma_trigger(struct snd_pcm_substream *substream, int cmd)
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_STOP:
-		ret = hda_link_dma_cleanup(substream, hext_stream, cpu_dai, codec_dai, true);
+		snd_hdac_ext_stream_clear(hext_stream);
+		ret = hda_link_dma_cleanup(substream, hext_stream, cpu_dai, codec_dai);
```

Comment:
I am pretty sure we added that 'trigger_stop' flag on purpose in the past to keep consistent sequences between different cases. I don't mind undoing this, but this wasn't a random decision.

Reply:
@plbossart whatever the reason for the split was, this is needed to unify the DMA trigger op for IPC3 and IPC4.

Reply:
> @plbossart whatever the reason for the split was, this is needed to unify the DMA trigger op for IPC3 and IPC4.

Is this a goal?

I thought the main goal was to unify the HDAudio-based DAI handling (existing, DSP less, future extensions). 

Do we want to unify the DMA trigger for IPC3 and IPC4? Or is this made possible by the refactoring?

A bit confused here.